File: Programmers/hash1.py
def solution(participant, completion):
    answer = ''
    participants = {}

    for p in participant:
        if p in participants.keys():
            participants[p] += 1
        else:
            participants[p] = 1

    for c in completion:
        participants[c] -= 1

    for p in participants:
        if participants[p] > 0:
            answer = p
            break

    return answer

participant = ["leo", "kiki", "eden"]
completion = ["eden", "kiki"]
# answer = "leo"
print("#1", solution(participant, completion))

participant = ["marina", "josipa", "nikola", "vinko", "filipa"]
completion = ["josipa", "filipa", "marina", "nikola"]
# answer = "vinko"
print("#2", solution(participant, completion))

participant = ["mislav", "stanko", "mislav", "ana"]
completion = ["stanko", "ana", "mislav"]
# answer = "mislav"
print("#3", solution(participant, completion))

File: Programmers/hash1.js
function solution(participant, completion) {
  var answer = "";

  const result = participant.reduce(
    (r, p) => ((r[p] = r[p] ? r[p] + 1 : 1), r),
    {}
  );
  completion.map((c) => result[c]--);

  for (let key in result) {
    if (result[key] > 0) {
      answer = key;
      break;
    }
  }

  return answer;
}

let participant = ["leo", "kiki", "eden"];
let completion = ["eden", "kiki"];
// answer = "leo"
console.log(solution(participant, completion));

participant = ["marina", "josipa", "nikola", "vinko", "filipa"];
completion = ["josipa", "filipa", "marina", "nikola"];
// answer = "vinko"
console.log(solution(participant, completion));

participant = ["mislav", "stanko", "mislav", "ana"];
completion = ["stanko", "ana", "mislav"];
// answer = "mislav"
console.log(solution(participant, completion));
